### ISSUE TITLE
[14.0] base_import_match: Unable to use a closed cursor.

### ISSUE THREAD

**Issue by dreispt** [OCA/server-backend#123]
I'm consistently getting this error.
The first import works. If I repeat it a second time, I get this error.

```
2021-04-26 16:43:52,121 31401 INFO 14-vin-0412 odoo.addons.base_import.models.base_import: importing 2 rows...
2021-04-26 16:43:52,647 31401 INFO 14-vin-0412 odoo.addons.base_import.models.base_import: done
2021-04-26 16:44:03,921 31401 INFO 14-vin-0412 odoo.addons.base_import.models.base_import: importing 2 rows...
2021-04-26 16:44:04,061 31401 ERROR 14-vin-0412 odoo.http: Exception during JSON request handling.
Traceback (most recent call last):                                                                                                                                                                                                             File "/home/daniel/work14/bot/odoo-boti/odoo/odoo/addons/base/models/ir_http.py", line 237, in _dispatch                                                                                                                                       result = request.dispatch()                                                                                                                                                                                                                File "/home/daniel/work14/bot/odoo-boti/odoo/odoo/http.py", line 683, in dispatch
    result = self._call_function(**self.params)
  File "/home/daniel/work14/bot/odoo-boti/odoo/odoo/http.py", line 359, in _call_function                                                                                                                                                        return checked_call(self.db, *args, **kwargs)
  File "/home/daniel/work14/bot/odoo-boti/odoo/odoo/service/model.py", line 94, in wrapper                                                                                                                                                       return f(dbname, *args, **kwargs)
  File "/home/daniel/work14/bot/odoo-boti/odoo/odoo/http.py", line 347, in checked_call                                                                                                                                                          result = self.endpoint(*a, **kw)
  File "/home/daniel/work14/bot/odoo-boti/odoo/odoo/http.py", line 912, in __call__
    return self.method(*args, **kw)
  File "/home/daniel/work14/bot/odoo-boti/odoo/odoo/http.py", line 531, in response_wrap                                                                                                                                                         response = f(*args, **kw)
  File "/home/daniel/work14/vin/vinha-odoo/odoo/addons/web/controllers/main.py", line 1377, in call_kw                                                                                                                                           return self._call_kw(model, method, args, kwargs)
  File "/home/daniel/work14/vin/vinha-odoo/odoo/addons/web/controllers/main.py", line 1369, in _call_kw                                                                                                                                          return call_kw(request.env[model], method, args, kwargs)
  File "/home/daniel/work14/bot/odoo-boti/odoo/odoo/api.py", line 396, in call_kw
    result = _call_kw_multi(method, model, args, kwargs)
  File "/home/daniel/work14/bot/odoo-boti/odoo/odoo/api.py", line 383, in _call_kw_multi                                                                                                                                                         result = method(recs, *args, **kwargs)
  File "/home/daniel/work14/vin/vinha-odoo/src/enterprise/account_bank_statement_import_csv/wizard/account_bank_statement_import_csv.py", line 177, in do                                                                                        
  return super(AccountBankStmtImportCSV, self).do(fields, columns, options, dryrun=dryrun)                                                                                                                                                   File "/home/daniel/work14/vin/vinha-odoo/odoo/addons/base_import/models/base_import.py", line 928, in do                                                                                                                                       import_result = model.load(import_fields, data)
  File "/home/daniel/work14/vin/vinha-odoo/src/custom-addons/base_import_match/models/base.py", line 45, in load                                                                                                                                 match = self.env["base_import.match"]._match_find(self, record, row)
  File "/home/daniel/work14/vin/vinha-odoo/src/custom-addons/base_import_match/models/base_import.py", line 84, in _match_find                                                                                                                   
    for field in combination.field_ids:
  File "/home/daniel/work14/bot/odoo-boti/odoo/odoo/fields.py", line 3144, in __get__
    return super().__get__(records, owner)
  File "/home/daniel/work14/bot/odoo-boti/odoo/odoo/fields.py", line 2483, in __get__
    return super().__get__(records, owner)
  File "/home/daniel/work14/bot/odoo-boti/odoo/odoo/fields.py", line 996, in __get__
    recs._fetch_field(self)
  File "/home/daniel/work14/bot/odoo-boti/odoo/odoo/models.py", line 3069, in _fetch_field                                                                                                                                                       self._read(fnames)
  File "/home/daniel/work14/bot/odoo-boti/odoo/odoo/models.py", line 3164, in _read
    field.read(fetched)
  File "/home/daniel/work14/bot/odoo-boti/odoo/odoo/fields.py", line 3155, in read
    lines = comodel.search(domain, limit=self.limit)
  File "/home/daniel/work14/bot/odoo-boti/odoo/odoo/models.py", line 1709, in search
    return res if count else self.browse(res)
  File "/home/daniel/work14/bot/odoo-boti/odoo/odoo/models.py", line 4954, in browse
    if not ids:
  File "/home/daniel/work14/bot/odoo-boti/odoo/odoo/osv/query.py", line 156, in __bool__
    return bool(self._result)
  File "/home/daniel/work14/bot/odoo-boti/odoo/odoo/tools/func.py", line 26, in __get__
    value = self.fget(obj)
  File "/home/daniel/work14/bot/odoo-boti/odoo/odoo/osv/query.py", line 149, in _result
    self._cr.execute(query_str, params)
  File "<decorator-gen-3>", line 2, in execute
  File "/home/daniel/work14/bot/odoo-boti/odoo/odoo/sql_db.py", line 100, in check
    raise psycopg2.OperationalError('Unable to use a closed cursor.')
Exception

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/daniel/work14/bot/odoo-boti/odoo/odoo/http.py", line 639, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/home/daniel/work14/bot/odoo-boti/odoo/odoo/http.py", line 315, in _handle_exception
    raise exception.with_traceback(None) from new_cause
psycopg2.OperationalError: Unable to use a closed cursor.
```

**Comment by yajo** [comment]
Please may you reformat the logs? I can't understand them as they are.

**Comment by dreispt** [comment]
Can you be more specific, is the problem scrolling right on the longer log lines?

**Comment by yajo** [comment]
Yes

**Comment by dreispt** [comment]
Closed by #134
